manufacturing-client/src/main.rs

```diff
@@ -295,11 +298,70 @@ async fn get_mfg_info(mfg_string_type: MfgStringType) -> Result<CborSimpleType>
         return Ok(CborSimpleType::Text(mfg_info.into_string().unwrap()));
     }
     match mfg_string_type {
-        MfgStringType::SerialNumber => {
-            let serial = fs::read_to_string("/sys/devices/virtual/dmi/id/product_serial")
+        MfgStringType::SerialNumberExt => {
+            let serial_dmi = fs::read_to_string("/sys/devices/virtual/dmi/id/product_serial")
                 .or_else(|_| fs::read_to_string("/sys/devices/virtual/dmi/id/chassis_serial"))
                 .context("Error determining system serial number")?;
-            Ok(CborSimpleType::Text(serial))
+            let mut arg: String;
+            let output_ifaces = Command::new("ls")
+                .arg("/sys/class/net/")
+                .stdout(std::process::Stdio::piped())
+                .stderr(std::process::Stdio::piped())
+                .output()
+                .expect("Failed to execute command");
+
+            let val = String::from_utf8_lossy(&output_ifaces.stdout).to_string();
+            let mut serial_mac: String = "".to_string();
+            for interface in val.split('\n') {
+                arg = "/sys/class/net/".to_owned() + interface + "/address";
+                let output = Command::new("/bin/cat")
+                    .args([":", &arg])
+                    .stdout(std::process::Stdio::piped())
+                    .stderr(std::process::Stdio::piped())
+                    .output()
+                    .expect("Failed to execute grep command");
+                let mac = String::from_utf8_lossy(&output.stdout)
+                    .trim_end()
+                    .to_string();
+                let mac_regex = Regex::new(r"^([0-9A-Fa-f]{2}[:-]){5}([0-9A-Fa-f]{2})$").unwrap();
```

Comment:
Often want to cache regexps; check out https://github.com/coreos/coreos-installer/pull/1101

Reply:
great catch !